src/main/java/teammates/sqllogic/core/UsersLogic.java

```diff
@@ -184,6 +184,22 @@ public List<Student> getStudentsForCourse(String courseId) {
         return studentReturnList;
     }
 
+    /**
+     * Gets a list of students for the specified course.
+     */
+    public List<Student> getUnregisteredStudentsForCourse(String courseId) {
+        List<Student> students = getStudentsForCourse(courseId);
+        List<Student> unregisteredStudents = new ArrayList<>();
+
+        for (Student s : students) {
+            if (s.getGoogleId() == null || s.getGoogleId().trim().isEmpty()) {
+                unregisteredStudents.add(s);
+            }
```

Comment:
```suggestion
            if (s.getAccount() == null) {
                unregisteredStudents.add(s);
            }
```

Checking for account should be sufficient